### PR TITLE
Basemap selector

### PR DESCRIPTION
**Pull request by JoaquinTrinanes**
This PR adds a toggle between default and satellite map.

One thing to note is that when switching to the default style the satellite layer is removed. When the same behaviour was implemented toggling it's visibility things started going weird: the map flickers, the tiles load again... I remove it for these reason.

### CODE REVIEW

**Review by davidsingal — CHANGES_REQUESTED** [review]
Screenshot 2022-06-01 at 10 32 38

The main impact layer is behind the satellite layer. The basemaps should be always under the layers.

File: client/src/components/map/controls/basemap/component.tsx [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+  }, [onChange, value]);
+
+  const image = value === 'default' ? SatelliteImage : DefaultImage;
+
+  return (
+    <Tooltip
+      hoverTrigger
+      placement="right"
+      theme="dark"
+      content={<div className="bg-gray-900 text-white rounded-lg p-2">Change basemap!</div>}
```
Comment (davidsingal) [inline client/src/components/map/controls/basemap/component.tsx:31]:
```suggestion
      content={<div className="bg-gray-900 text-white rounded-lg p-2">Change basemap</div>}
```

File: client/src/components/map/controls/basemap/component.tsx [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+      onChange('satellite');
+    } else {
+      onChange('default');
+    }
+  }, [onChange, value]);
+
+  const image = value === 'default' ? SatelliteImage : DefaultImage;
+
+  return (
+    <Tooltip
```
Comment (davidsingal) [inline client/src/components/map/controls/basemap/component.tsx:27]: Do we need a tooltip? Is it in the design??

File: client/src/components/tooltip/component.tsx [hunk 3]
```diff
@@ -38,34 +47,59 @@ export const ToolTip: React.FC<TooltipProps> = ({
...
-      </Popover.Button>
-      <Popover.Panel
-        ref={floating}
-        className={className}
-        style={{
-          position: strategy,
-          top: y ?? '',
-          left: x ?? '',
+      <Popover.Button
+        onMouseEnter={() => {
```
Comment (davidsingal) [inline client/src/components/tooltip/component.tsx:62]: consider `useCallback`

File: client/src/components/tooltip/component.tsx [hunk 4]
```diff
@@ -38,34 +47,59 @@ export const ToolTip: React.FC<TooltipProps> = ({
...
+          setIsHovered(true);
+        }}
+        onMouseLeave={() => {
```
Comment (davidsingal) [inline client/src/components/tooltip/component.tsx:65]: same, consider `useCallback`

**Review by davidsingal — CHANGES_REQUESTED** [review]

**Comment by davidsingal** [comment]
Also, it seems like the design changed a bit: https://www.figma.com/file/WxYMmJrUZAgV6f0Csmt8fK/LG-Tool---Work-in-progress?node-id=1724%3A40086

File: client/src/components/map/controls/basemap/component.tsx (hunk 1 above)
Comment (JoaquinTrinanes) [reply client/src/components/map/controls/basemap/component.tsx]: Removed it, also added the name of the basemap that will be changed to on swap

**Comment by JoaquinTrinanes** [comment]
> Also, it seems like the design changed a bit: https://www.figma.com/file/WxYMmJrUZAgV6f0Csmt8fK/LG-Tool---Work-in-progress?node-id=1724%3A40086

That explains why I couldn't find the design 🥲

File: client/src/components/map/controls/basemap/component.tsx (hunk 2 above)
Comment (JoaquinTrinanes) [reply client/src/components/map/controls/basemap/component.tsx:23]: I just found the new design, I'll ask Andreia if we still need that tooltip

File: client/src/components/tooltip/component.tsx (hunk 4 above)
Comment (JoaquinTrinanes) [reply client/src/components/tooltip/component.tsx]: It won't probably matter, but I rather be consistent

File: client/src/components/map/controls/basemap/component.tsx (hunk 2 above)
Comment (JoaquinTrinanes) [reply client/src/components/map/controls/basemap/component.tsx:23]: image
Can confirm. I'll change the text accordingly 😁

**Comment by davidsingal** [comment]
Screenshot 2022-06-06 at 11 15 52
Screenshot 2022-06-06 at 11 16 05

There is an issue with the tooltips (see screenshots).

**Review by davidsingal — CHANGES_REQUESTED** [review]
I think the issue is not with z-index. It seems like opacity 1 is still opacity, and that's why with a different background it looks weird. Could you check it?

**Review by davidsingal — APPROVED** [review]

**Pull request merged**